File: src/main/java/cookiedragon/luchadora/module/impl/render/FullBrightModule.java
package cookiedragon.luchadora.module.impl.render;

import cookiedragon.luchadora.event.api.Subscriber;
import cookiedragon.luchadora.event.client.UpdateLightmapEvent;
import cookiedragon.luchadora.module.AbstractModule;
import cookiedragon.luchadora.module.Category;
import cookiedragon.luchadora.value.values.NumberValue;

/**
 * @author cookiedragon234 01/Jan/2020
 */
@AbstractModule.Declaration(name = "Full Bright", description = "Get rid of darkness", category = Category.RENDER)
public class FullBrightModule extends AbstractModule
{
	private NumberValue brightness = new NumberValue("Brightness", 1f, 0, 3);
	
	public FullBrightModule()
	{
		brightness.addCallback((oldVal, newVal) ->
			updateLightmap(newVal.floatValue()));
	}
	
	@Subscriber
	private void onUpdateLightmap(UpdateLightmapEvent event)
	{
		updateLightmap(brightness.getValue().floatValue());
	}
	
	@Override
	protected void onEnabled()
	{
		updateLightmap(brightness.getValue().floatValue());
	}
	
	@Override
	protected void onDisabled()
	{
		mc.world.provider.generateLightBrightnessTable();
	}
	
	private static void updateLightmap(float brightness)
	{
		if (mc.world.nonNull())
		{
			for (int i = 0; i < mc.world.provider.getLightBrightnessTable().length; i++)
			{
				mc.world.provider.getLightBrightnessTable()[i] = brightness;
			}
		}
	}
}

File: src/main/kotlin/cookiedragon/luchadora/module/impl/render/FullBrightModule.kt
package cookiedragon.luchadora.module.impl.render

import cookiedragon.luchadora.event.api.Subscriber
import cookiedragon.luchadora.event.client.UpdateLightmapEvent
import cookiedragon.luchadora.module.AbstractModule
import cookiedragon.luchadora.module.Category
import cookiedragon.luchadora.util.Globals.mc
import cookiedragon.luchadora.value.values.NumberValue

/**
 * @author cookiedragon234 11/Jan/2020
 */
@AbstractModule.Declaration(name = "Full Bright", description = "Get rid of darkness", category = Category.RENDER)
class FullBrightModule : AbstractModule() {
	private val brightness = NumberValue("Brightness", 1f, 0, 3)
			.addCallback { oldVal, newVal -> updateLightmap(newVal.toFloat()) }
	
	@Subscriber
	private fun onUpdateLightmap(event: UpdateLightmapEvent) = updateLightmap(brightness.value.toFloat())
	
	override fun onEnabled() = updateLightmap(brightness.value.toFloat())
	
	override fun onDisabled() = mc.world.provider.generateLightBrightnessTable()
	
	private fun updateLightmap(brightness: Float) {
		if (mc.world.nonNull()) {
			for (i in 0 until mc.world.provider.lightBrightnessTable.size) {
				mc.world.provider.lightBrightnessTable[i] = brightness
			}
		}
	}
}
